File: day8/part2.py
# https://adventofcode.com/2015/day/8
import re

strings = open('input').read().splitlines()
# strings = open('example').read().splitlines()

strlen_original = 0
strlen_encoded = 0

for string in strings:
    strlen_original += len(string)
    # encoded = eval(string)
    encoded = '"' + re.sub(
        r'(["\\])',
        r'\\\1',
        string
    ) + '"'
    strlen_encoded += len(encoded)

strlen_diff = strlen_encoded - strlen_original

print('Diff: {}'.format(strlen_diff))

File: day8/part2.js
const fs = require('fs');

let lines = fs.readFileSync('input', {encoding: 'utf8'}).split(/\r?\n/);

let strlen_original = 0;
let strlen_encoded = 0;

for (let n in lines) {
    line = lines[n];
    strlen_original += line.length;
    encoded = '"' + line.replace(/(["\\])/g, '\\$1') + '"';
    strlen_encoded += encoded.length
}

console.log('Diff: ' + (strlen_encoded - strlen_original));
